guides/common/modules/proc_configuring-direct-ad-integration-with-gss-proxy.adoc

```diff
@@ -40,24 +40,50 @@ security = ads
 # id {apache-user}
 ----
 +
-Apache user must not have access to the keytab file.
+The Apache user must not have access to the keytab file.
 . Create the `/etc/gssproxy/00-http.conf` file with the following content:
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
 [service/HTTP]
 mechs = krb5
-cred_store = keytab:/etc/httpd/conf/http.keytab
+cred_store = keytab:/etc/gssproxy/http.keytab
 cred_store = ccache:/var/lib/gssproxy/clients/krb5cc_%U
 euid = __ID_of_Apache_User__
 ----
+. Add a SELinux context rule to apply the appropriate security context to the `http.keytab` file:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# semanage fcontext -a -s system_u -t krb5_keytab_t '/etc/gssproxy/http\.keytab'
+----
 . Create a keytab entry:
 +
 [options="nowrap", subs="+quotes,verbatim,attributes"]
 ----
-# KRB5_KTNAME=FILE:/etc/httpd/conf/http.keytab net ads keytab add HTTP -U administrator -d3 -s /etc/net-keytab.conf
-# chown root.{apache-user} /etc/httpd/conf/http.keytab
-# chmod 640 /etc/httpd/conf/http.keytab
+# KRB5_KTNAME=FILE:/etc/gssproxy/http.keytab net ads keytab add HTTP -U administrator -d3 -s /etc/net-keytab.conf
+# chown root.root /etc/gssproxy/http.keytab
+# chmod 600 /etc/gssproxy/http.keytab
+----
+. Configure SSSD to map Group Policy Objects (GPOs) from AD to the `foreman` PAM service:
+.. In your `/etc/sssd/sssd.conf` file, add the following lines to the `domain` section for the Active Directory domain:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+[domain/_EXAMPLE_]
+access_provider = ad
+ad_gpo_access_control = enforcing
+ad_gpo_map_service = +foreman
+----
+ifdef::satellite[]
++
+For more information on GPOs, see link:{RHELDocsBaseURL}9/html/integrating_rhel_systems_directly_with_windows_active_directory/managing-direct-connections-to-ad_integrating-rhel-systems-directly-with-active-directory#how-sssd-interprets-gpo-access-control-rules_applying-group-policy-object-access-control-in-rhel[How SSSD interprets GPO access control rules] in _{RHEL}{nbsp}9 Integrating RHEL systems directly with Windows Active Directory_ or link:{RHELDocsBaseURL}8/html/integrating_rhel_systems_directly_with_windows_active_directory/managing-direct-connections-to-ad_integrating-rhel-systems-directly-with-active-directory#applying-group-policy-object-access-control-in-rhel_managing-direct-connections-to-ad[How SSSD interprets GPO access control rules] in _{RHEL}{nbsp}8 Integrating RHEL systems directly with Windows Active Directory_.
+endif::[]
+.. Restart the `sssd` service:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# systemctl restart sssd
 ----
 . Enable IPA authentication in {Project}:
```

Comment:
Looking at this, I doubt it's really correct. That feature is really designed for IPA, not for AD.

For example, it ensures the keytab is owned by Apache, which directly contradicts the statement that Apache shouldn't be able to read the keytab. However, it does set up some things in Apache that may be needed.

Big question: what is actually needed for the AD integration?

Reply:
I believe in this case, AD is masked as IPA by SSSD. That's how I always imagined it to work.

Reply:
Further elaborating on this: we explicitly tell Apache about its location: https://github.com/theforeman/puppet-foreman/blob/5f38479c870ef03a4a1565007be68b61e8950d53/templates/auth_gssapi.conf.erb#L6

Reply:
It also inspired me to refactor a bit: https://github.com/theforeman/puppet-foreman/pull/1172

Reply:
> I believe in this case, AD is masked as IPA by SSSD. That's how I always imagined it to work.

I'd classify it as a hack and fundamentally there are some differences here that IMHO make it incompatible. In particular, the keytab handling is different. At least when I read this documentation and the code for IPA.

Reply:
Unaccessibility by Apache can still be achieved using permissions and selinux context on that file, right?

Reply:
Does Apache only need to read it during installation or does it access it regularly? If the latter, we either need to change the code or the goal outlined in the docs can't be reached IMO.

Reply:
Anyway, when I used path `/etc/httpd/config/http.conf` with `root.root`, `600` and `krb5_keytab_t`, it worked for me.

Reply:
```
# sealert -l 20295c58-da31-43c5-897c-63c5e83544e0
SELinux is preventing /usr/sbin/gssproxy from search access on the directory httpd.

*****  Plugin catchall (100. confidence) suggests   **************************

If you believe that gssproxy should be allowed search access on the httpd directory by default.
Then you should report this as a bug.
You can generate a local policy module to allow this access.
Do
allow this access for now by executing:
# ausearch -c 'gssproxy' --raw | audit2allow -M my-gssproxy
# semodule -X 300 -i my-gssproxy.pp


Additional Information:
Source Context                system_u:system_r:gssproxy_t:s0
Target Context                system_u:object_r:httpd_config_t:s0
Target Objects                httpd [ dir ]
Source                        gssproxy
Source Path                   /usr/sbin/gssproxy
Port                          <Unknown>
Host                         <FQDN>
Source RPM Packages           
Target RPM Packages           
SELinux Policy RPM            selinux-policy-targeted-3.14.3-139.el8_10.noarch
Local Policy RPM              selinux-policy-targeted-3.14.3-139.el8_10.noarch
Selinux Enabled               True
Policy Type                   targeted
Enforcing Mode                Enforcing
Host Name                     <FQDN>
Platform                      Linux
                              <FQDN>
                              4.18.0-553.8.1.el8_10.x86_64 #1 SMP Fri Jun 14
                              03:19:37 EDT 2024 x86_64 x86_64
Alert Count                   9
First Seen                    2024-07-25 05:10:52 EDT
Last Seen                     2024-07-25 09:12:22 EDT
Local ID                      20295c58-da31-43c5-897c-63c5e83544e0

Raw Audit Messages
type=AVC msg=audit(1721913142.873:1466): avc:  denied  { search } for  pid=27259 comm="gssproxy" name="httpd" dev="vda3" ino=159424064 scontext=system_u:system_r:gssproxy_t:s0 tcontext=system_u:object_r:httpd_config_t:s0 tclass=dir permissive=0


Hash: gssproxy,gssproxy_t,httpd_config_t,dir,search
```

Reply:
And the autogenerated rule is:
```
module my-gssproxy 1.0;

require {
        type httpd_config_t;
        type gssproxy_t;
        class dir search;
}

#============= gssproxy_t ==============
allow gssproxy_t httpd_config_t:dir search;
```
That looks like gssproxy just isn't supposed to look into Apache's folder. It all looks hacky.
Surprisingly, it works.

Reply:
With the latest commit, I went with @lhellebr's "hacky but works" approach.